group-coordinator/src/main/java/org/apache/kafka/coordinator/group/OffsetMetadataManager.java

```diff
@@ -539,26 +564,59 @@ public OffsetFetchResponseData.OffsetFetchResponseGroup fetchAllOffsets(
     }
 
     public void cleanupOffsetMetadata(List<Record> records, long offsetsRetentionMs) {
-        // TODO: get only committed offset groups?
         offsetsByGroup.forEach((groupId, offsetsByTopic) -> {
```

Comment:
This approach would not catch empty groups without offsets, isn't it? I think that we should take an approach similar to what we did for the delete group api. We should iterate over all the groups from the group metadata manager in the shard. For each group, we could clean the offsets and the remove the group if empty.

It is also a but weird to have the offset metadata manager delete groups.